stubs/Driver/Cursor.stub.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace MongoDB\Driver;
+
+/**
+ * @template-covariant TValue of array|object
+ *
+ * @template-implements CursorInterface<TValue>
+ */
+final class Cursor implements CursorInterface
+{
+    /**
+     * @return TValue|null
+     * @psalm-ignore-nullable-return
```

Comment:
IIUC, [`@psalm-ignore-nullable-return`](https://psalm.dev/docs/annotating_code/supported_annotations/#psalm-ignore-nullable-return) tells Psalm to never consider a null return value. But `current()` and `key()` can certainly return `null` if `valid()` is `false`. Why needn't we consider that case?

Reply:
This was blatantly copied over from [psalm's iterator stubs](https://github.com/vimeo/psalm/blob/d747f6500b38ac4f7dfc5edbcae6e4b637d7add0/stubs/CoreGenericIterators.phpstub#L27). The way I see it, this is a shortcoming of the original interface. The interface doesn't specify whether `current` and `key` should throw when on an invalid position, and most iterators return `null` instead. IMO, we should throw, but I imagine a lot of code isn't exactly built to handle throwing in the methods. At the same time, we do expect people to not call `current` and `key` without checking that the iterator is on a valid position, hence that annotation.

For immutable classes, the `psalm-assert-if-true` annotation would solve this, but since `next` has to reset a previously memoized result from `valid` I'm not sure if psalm can handle it. It would definitely be a worthwhile improvement.

Reply:
Thanks for the explanation.

I imagine throwing would be incredibly annoying, as manual iteration is already a complicated subject. I reckon the average PHP developer isn't familiar with the sequence of method calls to mimic a `foreach` loop.

Not so much the case with our write result getters relying on `isAcknowledged()`, as that's easier to wrap one's head around.